app/src/behaviors/behavior_sticky_key.c

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_sticky_key
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <zmk/behavior.h>
+
+#include <zmk/matrix.h>
+#include <zmk/endpoints.h>
+#include <zmk/event-manager.h>
+#include <zmk/events/position-state-changed.h>
+#include <zmk/events/keycode-state-changed.h>
+#include <zmk/events/modifiers-state-changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_NODE_EXISTS(DT_DRV_INST(0))
+
+#define ZMK_BHV_STICKY_KEY_MAX_HELD 10
+
+// increase this if you need more keys in the board
+#define ZMK_BHV_STICKY_KEY_POSITION_NOT_USED (-1)
+
+struct behavior_sticky_key_config {
+    s32_t release_after_ms;
+    struct zmk_behavior_binding behavior;
+};
+
+struct active_sticky_key {
+    s32_t position;
+    s32_t param1;
+    s32_t param2;
+    const struct behavior_sticky_key_config *config;
+    // timer data.
+    bool timer_started;
+    s32_t release_at;
+    struct k_delayed_work release_timer;
+    bool timer_is_cancelled;
+    // usage page and keycode for the key that is being modified by this sticky key
+    u8_t modified_key_usage_page;
+    u32_t modified_key_keycode;
+};
+
+struct active_sticky_key active_sticky_key[ZMK_BHV_STICKY_KEY_MAX_HELD] = {};
+
+static struct active_sticky_key *store_sticky_key(u32_t position, u32_t param1, u32_t param2,
+                                                  const struct behavior_sticky_key_config *config) {
+    for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
+        if (active_sticky_key[i].position != ZMK_BHV_STICKY_KEY_POSITION_NOT_USED) {
+            continue;
+        }
+        active_sticky_key[i].position = position;
+        active_sticky_key[i].param1 = param1;
+        active_sticky_key[i].param2 = param2;
+        active_sticky_key[i].config = config;
+        active_sticky_key[i].release_at = 0;
+        active_sticky_key[i].timer_is_cancelled = false;
+        active_sticky_key[i].timer_started = false;
+        active_sticky_key[i].modified_key_usage_page = 0;
+        active_sticky_key[i].modified_key_keycode = 0;
+        return &active_sticky_key[i];
+    }
+    return NULL;
+}
+
+static void clear_sticky_key(struct active_sticky_key *sticky_key) {
+    sticky_key->position = ZMK_BHV_STICKY_KEY_POSITION_NOT_USED;
```

Comment:
Any reason not to just use `memset` to clear this out entirely, and then set the position value?

Reply:
`    struct k_delayed_work release_timer;` needs to be setup and I think it's not supposed to be cleared.

Reply:
This works, and explanation makes total sense. Thanks!